### PR TITLE
SDK-152 Set debug mode at server creation time

### PR DESCRIPTION
**Pull request by AdamGrzybkowski**
Workflow described [here](https://issues.openmrs.org/browse/SDK-152)

### CODE REVIEW

**Comment by dkayiwa** [comment]
@rkorytkowski do you think this is ready for merging?

**Comment by dkayiwa** [comment]
@AdamGrzybkowski did you notice the merge conflicts on this?

**Comment by AdamGrzybkowski** [comment]
@dkayiwa thanks Daniel, conflicts resolved :) PR ready for code review and merge :D @tmarzeion @PawelGutkowski

**Comment by dkayiwa** [comment]
@AdamGrzybkowski did you take a look at the travis failure?

**Comment by AdamGrzybkowski** [comment]
@dkayiwa  I've restarted the build. It should be fine :)

**Comment by dkayiwa** [comment]
@rkorytkowski do you think this is ready for merging? :)

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/Setup.java [hunk 1]
```diff
@@ -199,16 +210,29 @@ public String setup(Server server, boolean isCreatePlatform, boolean isCopyDepen
...
 
+    private void setDebugPort(Server server) {
+        if (StringUtils.isBlank(debug) || wizard.checkYes(debug)) {
+            debug = wizard.promptForValueIfMissingWithDefault(
+                    ENABLE_DEBUGGING_DEFAULT_MESSAGE,
+                    server.getDebugPort(),
+                    "port number",
+                    "no debugging");
+        }
+        if(StringUtils.isNumeric(debug)){
```
Comment (PawelGutkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/Setup.java]: I don't like the fact that non numeric input for debug port is silently ignored, I guess we should do something about it, e.g. show information that non-numeric input is invalid and ask for port value again.

Comment (AdamGrzybkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/Setup.java]: Done

Comment (PawelGutkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/Setup.java]: Thanks! :hamburger:

**Pull request merged**